pkg/component/component.go

```diff
@@ -94,6 +94,19 @@ func CreateFromPath(client *occlient.Client, name string, ctype string, path str
 	return nil
 }
 
+// CreateFromSvcCatalog create new component from serviceCatalog
+func CreateFromSvcCatalog(client *occlient.Client, componentName string, componentType string, applicationName string) error {
+	labels := componentlabels.GetLabels(componentName, applicationName, true)
+	// save component type as label
+	labels[componentlabels.ComponentTypeLabel] = componentType
+	err := client.CreateServiceInstance(componentName, componentType, labels)
```

Comment:
`componentName` should be "namespaced" (prefixed with application name) to allow multiple ServiceCatalog components in different odo applications.